src/textual/document/_document.py

```diff
@@ -320,6 +342,40 @@ def line_count(self) -> int:
         """Returns the number of lines in the document."""
         return len(self._lines)
 
+    def get_index_from_location(self, location: Location) -> int:
```

Comment:
I can see why both of these would be useful. Would you mind writing a few tests for them?

Also, we have a policy of avoiding shorthand variable names where possible, so if you wouldn't mind renaming `col` -> `column`, `i` -> `line_index`, etc. that'd be appreciated too.